slatedb/src/mem_table_flush.rs

```diff
@@ -45,6 +45,7 @@ impl MemtableFlusher {
         self.manifest.refresh().await?;
         let mut wguard_state = self.db_inner.state.write();
         wguard_state.merge_remote_manifest(self.manifest.prepare_dirty()?);
+        self.db_inner.db_stats.l0_sst_count.set(wguard_state.state().core().l0.len() as i64);
```

Comment:
WDYT about updating the stats in `StateModifier::merge_remote_manifest` (db_state.rs) instead? We have `new_l0` there. I think setting the stat at that point should get us what we want.

Reply:
Ah, it looks like we have `CompactorState::merge_remote_manifest` as well. So you'd need to set the stats there, too (with merged_l0s.len() I think?).

Reply:
Is db reference available in `CompactorState`?

Reply:
Yea, that's pretty heavy. Let's not do that.

Ok, this looks like a good change as it is now. It seems to cover both the periodic polling path as well as manifest writes (which happen when L0s are updated).

I think the compactor side still needs to be addressed. It also has a `load_manifest` that should update the stat as well.